unofficial-api/node/commands/roomBrowser.js

```diff
@@ -1,9 +1,16 @@
 const Commands = require('./commands');
+const LobbySettings = require('./lobbySettings')
 const EVENTS = require('../events');
 
 class RoomBrowser extends Commands {
-	host(settings, solo=false) {
-		const command = solo ? "host solo room" : "host room"
+	host(settings, doNotValidate=false) {
+		let command
+		if(doNotValidate){
+			command = "host room"
+		}else{
+			LobbySettings.validate(settings)
+			command = settings.gameMode === "Solo" ? "host solo room" : "host room"
```

Comment:
you cant host solo room without validate?

Reply:
I also don't like adding LobbySettings here as a "dependency". I think adding the doNotValidate attribute to getSettings would be better